src/main/java/org/testng/internal/reflect/DataProviderMethodMatcher.java

```diff
@@ -0,0 +1,67 @@
+package org.testng.internal.reflect;
+
+import org.testng.ITestContext;
+import org.testng.ITestResult;
+
+import java.lang.reflect.Method;
+
+/**
+ * Checks the conformance as per data-provide specifications.
+ *
+ * @author <a href="mailto:nitin.matrix@gmail.com">Nitin Verma</a>
+ */
+public class DataProviderMethodMatcher implements MethodMatcher {
```

Comment:
Maybe, but they are sharing the same interface and the same attributes (at least `ArrayEndingMethodMatcher`, `DataProviderMethodMatcher` and `DirectMethodMatcher`).